java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -351,6 +359,14 @@ public Builder headerAuthenticator(CallHeaderAuthenticator headerAuthenticator)
       return this;
     }
 
+    /**
+     * Set the header handler.
+     */
+    public Builder headerHandler(ServerHeaderHandler headerHandler) {
```

Comment:
Can we rename this to propertyHeaderHandler if we decide to rename `ServerHeaderHandler` 
 to `ServerPropertyHeaderHandler`?

Reply:
As mentioned in another comment, the intent is to handle all headers. You can interpret them as properties, but I think the name is reflective of intent.